core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1794,20 +1811,29 @@ class Log(@volatile private var _dir: File,
     def shouldDelete(segment: LogSegment, nextSegmentOpt: Option[LogSegment]) = {
       if (diff - segment.size >= 0) {
         diff -= segment.size
+        info(s"Segment with base offset ${segment.baseOffset} will be deleted due to" +
```

Comment:
In addition, it might be useful to know the total log size. Maybe we could include `size - diff` as the size after deletion?